### ISSUE TITLE
How could I get the action of any time-step in one loop of simulation?

### ISSUE THREAD

**Issue by ZhouYFeng** [cliport/cliport#26]
Hi Mohit! Thanks for your nice work. I am trying to replace the agent with my designed neural network for the prediction of the agent action. Here, I am confused that if I could record (obs, action) pairs at every time-step in one loop of simulation depending on a sampling rate by modifying **env.step()** or **agent.act()** function.

1. First, I use the PyBullet to randomly generate dataset, just like what you did in cliport/cliport/demo.py. And I can save the observation and corresponding action using 'episode.append((obs, act, reward, info))'. 

![image](https://user-images.githubusercontent.com/111882855/228486148-08cd69c2-6155-4d23-aecf-18446058952f.png)

2. I check that it seems like that only when a block has been placed correctly, an action will be output by the agent, which means that only 6 or 7 steps (observation, action) are recorded in one trajectory. 

3. For my training dataset, I want to record the (observation, action) pairs that the agent performs along the whole simulation loop depending on a sampling rate. In your code, you have provided 'record videos' to save mp4 for one completed simulation. Therefore, observations at any time step can be sampled. I want to know that if there any way for me to record the action corresponding to the sampled observation, instead of only saving the last completed action? Specifically, in the process of putting the green block in the brown box, there is a first process of lifting the green block up to certain height, then translating the green block, and finally putting down the block in box. In this procedure, 100 observations/frames can be sampled from mp4, but the number of recorded actions from agent is only 6 or 7. Could I save the (observation, action) pair at some time-step? For instance, the green block is being raised by the robot to a certain height at an intermediate location.

Appreciate and look for your reply! :)

**Comment by ZhouYFeng** [comment]
@MohitShridhar Okay, Thanks for your reply! I will try it.

**Comment by ZhouYFeng** [comment]
@MohitShridhar Sorry for disturbing again. You mentioned that **(1) save additional actions while saving video frames**, but I am not very sure where or which I should modify to record the corresponding action with the frame. I have understood how [This](https://github.com/cliport/cliport/blob/47bf8298f6b664365ab042eb7ac0bcaf5df60254/cliport/environments/environment.py#L396) works, but have no idea if I can record the corresponding action in this **add_video_frame()**?

Thanks a lot!
